vsm

```diff
@@ -147,31 +149,55 @@ def process(state, signal, value):
     state.got_signal(signal, value)
 
 
-def run(state):
+def receive():
```

Comment:
Right, this is another issue I actually wanted to discuss with you.

I am aware that indeed we need to replace the send function in this way, but then how could the tests check the output?, The reason I didn't replace this function is because I was also using the value printed to stdout to check for the expected output in the IPC tests, otherwise we would also need a 'receive' end in the tests and I am not quite sure how to go through this .. Would Shane proposal of using zeromq subscriber/publisher help here?